src/my-bot.ts

```diff
@@ -1,38 +1,423 @@
+import * as firebase from 'firebase';
+import { Position } from './interfaces/position';
+import { ShipPlace } from './interfaces/shipPlace';
+import { Shot } from './interfaces/shot';
+
 export class MyBot {

```

Comment:
Just going to throw it out there - this class was so large, github wouldn't load it by default!

I think you could easily be thinking about how you could cut this up into separate classes.